### PR TITLE
Pause video when app goes in background

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: packages/video_player/lib/video_player.dart [hunk 1]
```diff
@@ -261,6 +265,37 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   }
 }
 
+class _VideoAppLifeCycleObserver extends WidgetsBindingObserver {
+  bool wasPlayingBeforePause = false;
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: Make private?

File: packages/video_player/lib/video_player.dart [hunk 2]
```diff
@@ -261,6 +265,37 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
...
+  void initialize() {
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.paused:
+        wasPlayingBeforePause = controller.value.isPlaying;
+        if (controller.value.initialized) controller.pause();
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: Please format single-statement if branches consistently.

Worse: if controller is not yet initialized, we lose the pause signal and may start playing while in background.

**Review by mravn-google — CHANGES_REQUESTED** [review]

File: packages/video_player/lib/video_player.dart (hunk 1 above)
Comment (sigurdm) [reply packages/video_player/lib/video_player.dart]: DoneDone

File: packages/video_player/lib/video_player.dart (hunk 2 above)
Comment (sigurdm) [reply packages/video_player/lib/video_player.dart]: I removed the conditional. The controller should take care of delaying the play message till after initialisation is done.

**Review by mravn-google — APPROVED** [review]
LGTM

**Pull request merged**